api/service/src/main/java/com/coreeng/supportbot/teams/PlatformTeamsService.java

```diff
@@ -62,19 +82,85 @@ void init() {
                     ));
                 users.add(user);
             }
+            fetchedGroupUsers.put(groupRef, users);
+        }
 
-            groupIdToUsers.put(t.groupRef(), users);
-            for (PlatformUser user : users) {
-                team.users().add(user);
-                user.teams().add(team);
+        groupIdToUsers.putAll(fetchedGroupUsers);
+        for (PlatformTeam team : teamByName.values()) {
+            for (String groupRef : team.groupRefs()) {
+                List<PlatformUser> users = groupIdToUsers.getOrDefault(groupRef, List.of());
+                team.users().addAll(users);
+                for (PlatformUser user : users) {
+                    user.teams().add(team);
+                }
             }
         }
 
         log.atInfo()
             .addArgument(teamByName::size)
             .addArgument(groupIdToUsers::size)
             .addArgument(usersByEmail::size)
-            .log("Finished fetching teams info. Teams({}), Groups({}), Users({})");
+            .addArgument(() -> Duration.between(start, Instant.now()))
+            .log("Finished fetching teams info. Teams({}), Groups({}), Users({}), Elapsed({})");
+    }
+
+    private Map<String, List<PlatformUsersFetcher.Membership>> fetchMembershipsInParallel(
+        Set<String> groupRefs,
+        int maxConcurrency,
+        @Nullable Duration timeout
+    ) {
+        Map<String, CompletableFuture<List<PlatformUsersFetcher.Membership>>> futures = new HashMap<>();
+        if (groupRefs.isEmpty()) {
+            return new HashMap<>();
```

Comment:
Is it worth logging at this point? Could be useful for debugging team issues when groupRefs are returning empty